src/main/java/org/opensearch/neuralsearch/processor/combination/dto/CombineScoresDTO.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.neuralsearch.processor.combination.dto;
+
+import java.util.List;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NonNull;
+import org.apache.lucene.search.Sort;
+import org.opensearch.common.Nullable;
+import org.opensearch.neuralsearch.processor.CompoundTopDocs;
+import org.opensearch.neuralsearch.processor.combination.ScoreCombinationTechnique;
+import org.opensearch.search.query.QuerySearchResult;
+
+/**
+ * CombineScoresDTO handles the request sent for score combination after normalization.
+ */
+@AllArgsConstructor
+public class CombineScoresDTO {
```

Comment:
Please make it in a way the clients are "highly motivated" to use it in a builder fashion. Check out how it's done in say knn repo got [BaseQueryFactory](https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/index/query/BaseQueryFactory.java).
This is how [client use it](https://github.com/opensearch-project/k-NN/blob/main/src/main/java/org/opensearch/knn/index/query/KNNQueryFactory.java#L48-L56)